### ISSUE TITLE
[Accessibility] [Commerce Navigation] Although the menu behaves as a modal dialog, it does not have the proper markup or keyboard support for a dialog structure.

### ISSUE THREAD

**Issue by LevelAccessA11y** [adobe/aem-core-wcm-components#1789]
**Steps to Reproduce**
1. Enable NVDA screen reader.
2. Open page in Google Chrome.
3. Use TAB or TAB + SHIFT to navigate to the menu dialog trigger.
4. Activate menu dialog.
5. Use TAB or TAB + SHIFT to navigate through the dialog.
6. Notice the lack of titles and the keyboard focus moving beyond dialog content.

**Current Behavior**
Although the commerce navigation functions as a modal dialog, with a modal overlay obscuring the rest of the page content, the dialog lacks any proper role or state information communicated via `role` and `aria-` attributes.

Additionally, keyboard focus is not managed to stay within the dialog.

Screen reader users are not informed of the structure of the content, and sighted keyboard-only users may find their keyboard focus landing on content that is obscured by the modal overlay.

**Expected behavior/code**
Ensure that dialogs have proper markup and keyboard support by following the WAI Authoring Practices design pattern for [Dialog (Modal)](https://www.w3.org/TR/wai-aria-practices-1.1/#dialog_modal). 

**Environment**
Chrome 92, Windows 10, NVDA 2020.4 
AEM Core Components v. 2.17  

**Affected Users**
Keyboard-only users, screen reader users

**WCAG SC**
1.3.1 Info and Relationships
2.1.1 Keyboard

**Comment by richardhand** [comment]
@adobe export issue to Jira project SITES

**Comment by mhaack** [comment]
This applies to V1 of the navigation component which is not used any more.
